packages/rxtranscribe/src/operators/bufferBetweenSilences.js

```diff
@@ -10,29 +10,30 @@
 // }) {
 //   return chunk$ => {
 //     const chunkSub$ = chunk$.pipe(
-//       tap(c => console.log('CHUNK', c)),
+//       // tap(c => console.log('CHUNK', c)),
 //       share()
 //     );
 //     const chunkAndVad$ = chunkSub$.pipe(
-//       tap(c => console.log('CHUNK_AGAIN', c)),
+//       // tap(c => console.log('CHUNK_AGAIN', c)),
 //       _toVAD({...vadOptions}),
 //       share()
 //     );
 //     // close the buffer when both
 //     // (1) the time threshhold is exceeded and
 //     // (2) the VAD detects silence (a pause in the conversation)
 //     const stop$ = merge(of(true), interval(timeInterval)).pipe(
-//       tap(i => console.log('INTERVAL', i)),
+//       // tap(i => console.log('INTERVAL', i)),
 //       mergeMap(() => chunkAndVad$.pipe(
-//         tap(c => console.log('CHUNK_AND_VAD', c)),
+//         // tap(c => console.log('CHUNK_AND_VAD', c)),
 //         filter(([,vad]) => vad.type === 'SILENCE')
 //       )),
-//       tap(() => console.log('STOP'))
+//       // tap(() => console.log('STOP'))
 //     );
 //     const buffer$ = chunkSub$.pipe(
 //       // immediately emit the buffer when the stop$ condition is triggered
 //       bufferToggle(stop$, () => stop$),
-//       tap(b => console.log('BUFFER_OUT', b))
+//       mergeAll(),
+//       // tap(b => console.log('BUFFER_OUT', b))
 //     );
 //     return buffer$;
 //   };
```

Comment:
🤔 Cool.  I've never seen `mergeAll` before

Reply:
Uh wow. `mergeAll` is great.  I can't believe I missed that one.  The fact that you can control concurrency with it is great

Reply:
I had no issues getting it installed and running, but I couldn't get it working properly. There were no improvements in my workflow when using it despite numerous efforts. The `VAD` tests passed when I had it uncommented and ran them. The only difference it was making, was that it would ignore the first chunk of data causing me to miss the first sentence.

Reply:
I've been debugging `bufferOnSilence` for a couple hours now trying to get it to work with `GCP STT` and I think I figured out the problem. `VAD` works best on audio buffers between 26-144ms, yet we are feeding it chunks that are 25-150 times that size! We are effectively having VAD perform a useless operation. This means that the `bufferOnSilence` needs to take whatever chunk it was fed, chop off the last couple seconds and feed it into VAD until we have `x` milliseconds of consecutive silence and then return that partitioned chunk while simultaneously re-buffering the remaining amount so it can be prepended to the next incoming chunk. In the case of no more incoming chunks the last bit of data that was supposed to be prepended to the next incoming chunk is emitted.

Reply:
Oh... That explains it!